### PR TITLE
Add math notation to new functions

### PR DESCRIPTION
**Pull request by nerolation**

### CODE REVIEW

File: eip-5564.md [hunk 1]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+* User has access to the viewing private key (viewing key), spending private key $p_{spend}$.
+* User has access to a set of `Announcements` for which the `checkStealthAddress` function returns `true` . 
+- The  `computeStealthKey` function performs the following computations:
+  - Shared secret $s$ is computed by multiplying the viewing private key with the ephemeral public key of the announcement $s = p_{view}$ * $P_{ephemeral}$.
+  - The stealth private key is computed as $P_{stealth} = p_{spend} + s$.
+)
+
+---
 
 Sending funds now works as follows:
```
Comment (nerolation) [inline eip-5564.md]: I was not sure if we want to keep that section. It summarizes the above in a few lines, though, it might not be necessary. 
What do you think, @mds1 ?

File: eip-5564.md [hunk 2]
```diff
@@ -55,11 +55,13 @@ A recipient MUST be able to check if a stealth address belongs to them by callin
...
 /// @param ephemeralPubKey The ephemeral public key used to generate the stealth address.
 /// @param viewingKey The recipient's viewing private key.
+/// @param spendingPubKey The recipient's spending public key.
 /// @return True if funds sent to the stealth address belong to the recipient.
 function checkStealthAddress(
   address stealthAddress,
   bytes memory ephemeralPubKey,
-  bytes memory viewingKey
+  bytes memory viewingKey,
+  bytes memory spendingPubKey
```
Comment (nerolation) [inline eip-5564.md:64]: I added the spendingPubKey function which is required in the last step to derive the stealth address.

**Comment by nerolation** [comment]
yo @mds1, what do you think?

I added a mathematical describtion of the 3 functiones used to generate/verify a stealth address and derive the corresponding private key.

File: eip-5564.md [hunk 3]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
 ```
 
-An example stealth address scheme for elliptic curves is below. Other stealth schemes may need to modify this approach. **{TODO Modify these steps to match the three function signatures above}**
+An example stealth address scheme for elliptic curves is below. Other stealth schemes may need to modify this approach. 
 
-- $G$ is the generator point of the curve.
+- $G$ represents the generator point of the curve.
 
-- Recipient has private keys $p_{view}$ and $p_{spend}$.
+- Recipient has published a stealth meta-address that constists of the public keys $P_{spend}$ and $P_{view}$
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
- Recipient has published a stealth meta-address that consists of the public keys $P_{spend}$ and $P_{view}$
```

File: eip-5564.md [hunk 4]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
 
-- Recipient publishes corresponding public keys $P_{view}$ and $P_{spend}$ in the `IERC5564Registry`.
+- Recipient has has access to the private keys $p_{spend}$, $p_{view}$ that can access $P_{spend}$ and $P_{view}$.
```
Comment (mds1) [inline eip-5564.md]: Maybe this is a bit more precise?
```suggestion
- Recipient has has access to the private keys $p_{spend}$, $p_{view}$ from which $P_{spend}$ and $P_{view}$ are derived.
```

File: eip-5564.md [hunk 5]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
 
 - Sender generates random 32-byte entropy ephemeral private key $p_{ephemeral}$.
 
-- Sender passes the recipient address and $p_{ephemeral}$ to the `IERC5564Generator` contract's `generateStealthAddress` function.
+- Sender passes the stealth mata-address to the `generateStealthAddress` function.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
- Sender passes the stealth meta-address to the `generateStealthAddress` function.
```

File: eip-5564.md [hunk 6]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
 
-- This function performs the following computations:
+- The `generateStealthAddress` function performs the following computations:
+  - Generate a random ephemeral private key $p_{ephemeral}$.
```
Comment (mds1) [inline eip-5564.md]: This was already mentioned above, perhaps we delete that bullet (L131) and keep this one?

File: eip-5564.md [hunk 7]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+  - Derive the ephemeral public key  $P_{ephemeral}$ from $p_{ephemeral}$.
+  - Parse the spending and viewing public keys, P_{spend} and P_{view}, from the stealth meta-address.
   - A shared secret $s$ is computed as $s = p_{ephemeral} \cdot P_{view}$.
   - The secret is hashed $s_{h} = h(s)$.
```
Comment (mds1) [inline eip-5564.md]: Nit, since `h` is a function here we don't want it in italics
```suggestion
  - The secret is hashed $s_{h} = \textrm{h}(s)$.
```

File: eip-5564.md [hunk 8]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
 - Sender generates random 32-byte entropy ephemeral private key $p_{ephemeral}$.
 
-- Sender passes the recipient address and $p_{ephemeral}$ to the `IERC5564Generator` contract's `generateStealthAddress` function.
+- Sender passes the stealth mata-address to the `generateStealthAddress` function.
 
-- This function performs the following computations:
+- The `generateStealthAddress` function performs the following computations:
+  - Generate a random ephemeral private key $p_{ephemeral}$.
+  - Derive the ephemeral public key  $P_{ephemeral}$ from $p_{ephemeral}$.
+  - Parse the spending and viewing public keys, P_{spend} and P_{view}, from the stealth meta-address.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - Parse the spending and viewing public keys, $P_{spend}$ and $P_{view}$, from the stealth meta-address.
```

File: eip-5564.md [hunk 9]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
   - A shared secret $s$ is computed as $s = p_{ephemeral} \cdot P_{view}$.
   - The secret is hashed $s_{h} = h(s)$.
-  - The view tag $v$ is extracted by taking the most significant 12 bytes $s_{h}[0:12]$,
-  - Multiplying the shared secret with the generator point $S = s \cdot G$.
+  - The view tag $v$ is extracted by taking the most significant byte $s_{h}[0]$,
+  - Multiply the shared secret with the generator point $S = s \cdot G$.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - Multiply the hashed shared secret with the generator point $S_h = s_h \cdot G$.
```

File: eip-5564.md [hunk 10]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
   - The recipient's stealth public key is computed as $P_{stealth} = P_{spend} + S$.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - The recipient's stealth public key is computed as $P_{stealth} = P_{spend} + S_h$.
```

File: eip-5564.md [hunk 11]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
   - The recipient's stealth address $a_{stealth}$ is computed as $\textrm{pubkeyToAddress}(P_{stealth})$.
+  - The function returns the stealth address $a_{stealth}$, the ephemeral public key $P_{ephemeral}$ and the view tag $v$. 
+
+
+An example function to check whether a stealth address belongs to oneself:
+
+* User has access to the viweing private key $p_{view}$ and the public spending key $P_{spend}$.
+
+* User has access to a set of `Announcements` (ephemeral public keys) and applies the `checkStealthAddress` function to each of them.
```
Comment (mds1) [inline eip-5564.md]: There's more to an announcement log and it's already defined above, so maybe rephrase to this?
```suggestion
* User has access to a set of `Announcement` events and applies the `checkStealthAddress` function to each of them. 
```

File: eip-5564.md [hunk 12]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+
+An example function to check whether a stealth address belongs to oneself:
+
+* User has access to the viweing private key $p_{view}$ and the public spending key $P_{spend}$.
+
+* User has access to a set of `Announcements` (ephemeral public keys) and applies the `checkStealthAddress` function to each of them. 
+
+- The  `checkStealthAddress` function performs the following computations:
+  - Shared secret $s$ is computed by multiplying the viewing private key with the ephemeral public key of the announcement $s = p_{view}$ * $P_{ephemeral}$.
+  - Multiply the shared secret with the generator point $S = s \cdot G$.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - The secret is hashed $s_{h} = h(s)$.
  - Multiply the hashed shared secret with the generator point $S_h = s_h \cdot G$.
```

File: eip-5564.md [hunk 13]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+  - The stealth public key is computed as $P_{stealth} = P_{spend} + S$.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - The stealth public key is computed as $P_{stealth} = P_{spend} + S_h$.
```

File: eip-5564.md [hunk 14]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+  - The derived stealth address $a_{stealth}$ is computed as $\textrm{pubkeyToAddress}(P_{stealth})$.
+  - return `true` if the stealth address of the announcement matches the derived stealth address, else return `false`.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - Return `true` if the stealth address of the announcement matches the derived stealth address, else return `false`.
```

File: eip-5564.md [hunk 15]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+)
```
Comment (mds1) [inline eip-5564.md]: Stray parentheses?
```suggestion
```

File: eip-5564.md [hunk 16]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+
+An example function to derive the private key of a stealth address:
+* User has access to the viewing private key (viewing key), spending private key $p_{spend}$.
+* User has access to a set of `Announcements` for which the `checkStealthAddress` function returns `true` .
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
* User has access to a set of `Announcement` events for which the `checkStealthAddress` function returns `true` . 
```

File: eip-5564.md [hunk 17]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+  - The derived stealth address $a_{stealth}$ is computed as $\textrm{pubkeyToAddress}(P_{stealth})$.
+  - return `true` if the stealth address of the announcement matches the derived stealth address, else return `false`.
+)
+
+An example function to derive the private key of a stealth address:
+* User has access to the viewing private key (viewing key), spending private key $p_{spend}$.
+* User has access to a set of `Announcements` for which the `checkStealthAddress` function returns `true` . 
+- The  `computeStealthKey` function performs the following computations:
+  - Shared secret $s$ is computed by multiplying the viewing private key with the ephemeral public key of the announcement $s = p_{view}$ * $P_{ephemeral}$.
+  - The stealth private key is computed as $P_{stealth} = p_{spend} + s$.
```
Comment (mds1) [inline eip-5564.md]:
```suggestion
  - The secret is hashed $s_{h} = h(s)$.
  - The stealth private key is computed as $P_{stealth} = p_{spend} + s_h$.
```

File: eip-5564.md [hunk 18]
```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
...
+)
```
Comment (mds1) [inline eip-5564.md]: Stray parentheses again?
```suggestion
```

File: eip-5564.md (hunk 1 above)
Comment (mds1) [reply eip-5564.md]: I agree, I think we can remove this in favor of the above that uses the specific EIP methods
